design/nfs/nfs-controller-runtime.md

```diff
@@ -0,0 +1,161 @@
+# Implement controller-runtime in Rook NFS Operator
+
+## Background
+
+This proposal is to implement controller-runtime in Rook NFS Operator to improve reliability of the operator itself. Currently, Rook nfs-operator only simply watches an event of CustomResource from an informer using simple WatchCR method which has limited functionality such as event can not be re-queued if failed. To implement controller-runtime is expected to overcome the shortcomings of current implementation.
+
+## Why controller-runtime?
+
+Controller-runtime is widely used for writing Kubernetes operators. It is also leveraged by Kubebuilder and Operator SDK. Controller-runtime consists of several packages that have their respective responsibilities in building operators. The main function of controller-runtime is
```

Comment:
Let's link this first mention in this paragraph of controller-runtime to it's source repo, for easy access/context.

https://github.com/kubernetes-sigs/controller-runtime